forc-pkg/src/pkg.rs

```diff
@@ -406,7 +429,7 @@ fn remove_deps(
             .is_none()
             || to_remove
                 .iter()
-                .any(|removed_dep| removed_dep.1 == graph[node].unpinned(path_map))
+                .any(|removed_dep| *removed_dep == graph[node].unpinned(path_map).name)
```

Comment:
Are you sure it's enough to only check the `name` here? Keep in mind that a dependency graph can contain two instances of the same package at different versions or branches. Perhaps we should re-add the `Pkg` field to the `to_remove` list here?

Reply:
Hmm, that might be a little problematic to implement. Since the dependency is removed from the manifest file, we do not have any way of converting `Pinned` (which we can get from the `graph` that is created from the lock file) to `Pkg` so for removed packages I couldn't find a way to obtain their packages.

Reply:
Whether or not we use `Pkg`, the problem remains the same: by only checking packages by name, we will potentially remove unrelated, valid packages.

Can we perhaps pass through the node index or pinned ID instead? If not, we need to rethink how to approach this removal.